packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -257,106 +196,178 @@ fn eval_file<'s>(scope: &mut v8::HandleScope<'s>, path: &str) -> Result<Value<'s
+
+                        Error::PackageImport(
+                            path.to_string(),
+                            format!("Could not instantiate code for package: {exception_string}"),
+                        )
+                    })?;
+
+                if module.get_status() != v8::ModuleStatus::Instantiated {
+                    return Err(Error::PackageImport(
+                        path.to_string(),
+                        "Could not instantiate code for package".to_string(),
+                    ));
+                }
+
+                module.evaluate(&mut try_catch).ok_or_else(|| {
+                    let exception = try_catch.exception().unwrap();
+                    let exception_string = exception
+                        .to_string(&mut try_catch)
+                        .unwrap()
+                        .to_rust_string_lossy(&mut try_catch);
+
+                    Error::PackageImport(
+                        path.to_string(),
+                        format!("Could not evaluate code for package: {exception_string}"),
+                    )
+                })?;
+
+                // `v8::Module::evaluate` can return `Some` even though the evaluation didn't
+                // succeed
+                if module.get_status() != v8::ModuleStatus::Evaluated {
+                    let exception = module.get_exception();
+                    let exception_string = exception
+                        .to_string(&mut try_catch)
+                        .unwrap()
+                        .to_rust_string_lossy(&mut try_catch);
+
+                    return Err(Error::PackageImport(
+                        path.to_string(),
+                        format!("Could not evaluate code for package: {exception_string}"),
+                    ));
+                }
+
+                this.borrow_mut()
+                    .0
+                    .insert(path.to_string(), v8::Global::new(&mut try_catch, module));
+
+                Ok(Some(module))
+            }
+        }
+    }
+}
+
+// Simple example: https://gist.github.com/surusek/4c05e4dcac6b82d18a1a28e6742fc23e
+// More elaborate example: https://github.com/denoland/deno/blob/f7e7f548499eff8d2df0872d1340ddcdfa028c45/core/bindings.rs#L1344
+pub fn module_resolve_callback<'s>(
+    context: v8::Local<'s, v8::Context>,
+    // path of the module trying to get imported
+    specifier: v8::Local<'s, v8::String>,
+    _import_assertions: v8::Local<'s, v8::FixedArray>,
+    _referrer: v8::Local<'s, v8::Module>,
+) -> Option<v8::Local<'s, v8::Module>> {
+    // Safe: we are in a callback
+    let mut scope = unsafe { v8::CallbackScope::new(context) };
+    let specifier = specifier.to_rust_string_lossy(&mut scope);
+    ModuleMap::import_module(
+        scope
+            .get_slot::<Rc<RefCell<ModuleMap>>>()
+            .expect("ModuleMap is not present in isolate slots")
+            .clone(),
+        &mut scope,
+        &specifier,
+    )
+    .ok()
+    .flatten()
```

Comment:
I wanted to check and make sure the exception is actually helpful.
Apparently on missing dependency it's doing a core dump and not a pretty-ish exception. I might be missing an exception check somewhere.

So I added a `tracing::error` to explain there was an error when trying to import a dependency.
This also might be related to the fact that I wanted imports to stay simple. If/when we move towards something like deno, this might be fixed.